ToDoLite/src/main/java/com/couchbase/todolite/Application.java

```diff
@@ -298,11 +210,34 @@ public void onCreate() {
 
     }
 
-
     public Database getDatabase() {
         return this.database;
     }
 
+    public Database setDatabaseForName(String name) {
+        MessageDigest digest = null;
+        try {
+            digest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+        try {
+            byte[] inputBytes = name.getBytes();
+            byte[] hashBytes = digest.digest(inputBytes);
+            database = manager.getDatabase(byteArrayToHex(hashBytes));
```

Comment:
IIRC, db name cannot begin with number so to make sure that the db name is valid, you will need to prefix the name with some characters such as 'db'.